### ISSUE TITLE
Support widget tests

### ISSUE THREAD

**Issue by TobiasSuhrborg** [madewithfelt/flutter_layout_grid#3]
I wanted to have a widget test for your example, but it does not work.
I copied LayoutGridExample from the [example](https://github.com/madewithfelt/flutter_layout_grid/blob/master/example/basic.dart).
Note: LayoutGridExample is wrapped in a MaterialApp Widget because it needed a TextDirection to work in the widget test.

` testWidgets("test LayoutGridExample", (WidgetTester tester) async {
          await tester.pumpWidget(MaterialApp(
          home: Scaffold(body: LayoutGridExample()),
    ));
  });` 

The result was:
══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown during performLayout():
The getter 'minWidth' was called on null.
Receiver: null
Tried calling: minWidth

The relevant error-causing widget was:
  LayoutGrid
When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      RenderLayoutGrid.computeMinIntrinsicWidth (package:flutter_layout_grid/src/rendering/layout_grid.dart:193:33)
#2      RenderBox._computeIntrinsicDimension (package:flutter/src/rendering/box.dart:1299:20)
#3      RenderBox.getMinIntrinsicWidth (package:flutter/src/rendering/box.dart:1341:12)
#4      RenderBox.debugAssertDoesMeetConstraints.<anonymous closure>.testIntrinsic (package:flutter/src/rendering/box.dart:1986:41)
#5      RenderBox.debugAssertDoesMeetConstraints.<anonymous closure>.testIntrinsicsForValues (package:flutter/src/rendering/box.dart:1997:43)
#6      RenderBox.debugAssertDoesMeetConstraints.<anonymous closure> (package:flutter/src/rendering/box.dart:2004:32)
#7      RenderBox.debugAssertDoesMeetConstraints (package:flutter/src/rendering/box.dart:2028:6)
#8      RenderBox.size=.<anonymous closure> (package:flutter/src/rendering/box.dart:1745:7)
#9      RenderBox.size= (package:flutter/src/rendering/box.dart:1747:6)
#10     RenderLayoutGrid.performLayout (package:flutter_layout_grid/src/rendering/layout_grid.dart:257:9)
#11     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#12     RenderPadding.performLayout (package:flutter/src/rendering/shifted_box.dart:206:11)
#13     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#14     RenderConstrainedBox.performLayout (package:flutter/src/rendering/proxy_box.dart:259:13)
#15     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#16     MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:163:11)
#17     _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:477:7)
#18     MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:232:7)
#19     RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:391:14)
#20     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#21     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#22     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#23     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#24     _RenderCustomClip.performLayout (package:flutter/src/rendering/proxy_box.dart:1232:11)
#25     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#26     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#27     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#28     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#29     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#30     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#31     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#32     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#33     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#34     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#35     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#36     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#37     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#38     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#39     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#40     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#41     RenderOffstage.performLayout (package:flutter/src/rendering/proxy_box.dart:3168:13)
#42     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#43     RenderStack.performLayout (package:flutter/src/rendering/stack.dart:505:15)
#44     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#45     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#46     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#47     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#48     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#49     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#50     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#51     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#52     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#53     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#54     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#55     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#56     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#57     RenderProxyBoxMixin.performLayout (package:flutter/src/rendering/proxy_box.dart:105:13)
#58     RenderObject.layout (package:flutter/src/rendering/object.dart:1724:7)
#59     RenderView.performLayout (package:flutter/src/rendering/view.dart:167:13)
#60     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1584:7)
#61     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:844:18)
#62     AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:977:23)
#63     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:283:5)
#64     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1102:15)
#65     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1041:9)
#66     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:872:9)
#69     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#70     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:859:27)
#71     WidgetTester.pumpWidget.<anonymous closure> (package:flutter_test/src/widget_tester.dart:323:22)
#74     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#75     WidgetTester.pumpWidget (package:flutter_test/src/widget_tester.dart:320:27)
#76     main.<anonymous closure> 
#79     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:124:25)
#81     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:122:9)
#82     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:696:19)
#96     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1066:17)
#98     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1054:35)
(elided 35 frames from class _FakeAsync, package dart:async, package dart:async-patch, and package stack_trace)

The following RenderObject was being processed when the exception was fired: RenderLayoutGrid#09ebd NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE:
  creator: LayoutGrid ← Padding ← SizedBox.expand ← LayoutGridExample ← _BodyBuilder ← MediaQuery ←
    LayoutId-[<_ScaffoldSlot.body>] ← CustomMultiChildLayout ← AnimatedBuilder ← DefaultTextStyle ←
    AnimatedDefaultTextStyle ← _InkFeatures-[GlobalKey#fa485 ink renderer] ← ⋯
  parentData: offset=Offset(64.0, 0.0) (can use size)
  constraints: BoxConstraints(w=736.0, h=600.0)
  size: Size(736.0, 600.0)
This RenderObject had the following descendants (showing up to depth 5):
    child 1: RenderDecoratedBox#c35ca NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
      child: RenderPadding#70ded NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
        child: RenderLimitedBox#fcc45 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
          child: RenderConstrainedBox#52cb1 NEEDS-LAYOUT NEEDS-PAINT
    child 2: RenderDecoratedBox#fa04b NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
      child: RenderPadding#2166b NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
        child: RenderLimitedBox#201b5 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
          child: RenderConstrainedBox#72513 NEEDS-LAYOUT NEEDS-PAINT
    child 3: RenderDecoratedBox#c144d NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
      child: RenderPadding#6bf68 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
        child: RenderLimitedBox#a7c7d NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
          child: RenderConstrainedBox#1cc60 NEEDS-LAYOUT NEEDS-PAINT
    child 4: RenderDecoratedBox#4c85b NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
      child: RenderPadding#f23f0 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
        child: RenderLimitedBox#fc6db NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
          child: RenderConstrainedBox#f6952 NEEDS-LAYOUT NEEDS-PAINT
    child 5: RenderDecoratedBox#db41d NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
      child: RenderPadding#343cb NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
        child: RenderLimitedBox#8bbf9 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
          child: RenderConstrainedBox#f95e3 NEEDS-LAYOUT NEEDS-PAINT
    child 6: RenderDecoratedBox#2f451 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
      child: RenderPadding#ea346 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
        child: RenderLimitedBox#49d0c NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
          child: RenderConstrainedBox#74d25 NEEDS-LAYOUT NEEDS-PAINT
    child 7: RenderDecoratedBox#eb805 NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
      child: RenderPadding#3e3bd NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
        child: RenderLimitedBox#2e62e NEEDS-LAYOUT NEEDS-PAINT NEEDS-COMPOSITING-BITS-UPDATE
          child: RenderConstrainedBox#7d4d6 NEEDS-LAYOUT NEEDS-PAINT
════════════════════════════════════════════════════════════════════════════════════════════════════
This may be related to [another issue](https://github.com/madewithfelt/flutter_layout_grid/issues/1)

**Comment by shyndman** [comment]
Yes, it appears you're right.

Thanks for the report. I'll get that fixed up.

**Comment by shyndman** [comment]
I have a reproduction. Let's see if I can get this fixed. Have to get my brain back in grid mode.

**Comment by shyndman** [comment]
I got some basic intrinsic size functions in there. Probably not right exactly, but right enough to unblock you.

Published as v0.6.2

**Comment by shyndman** [comment]
I also fixed up the demos. Somewhere along the way they started failing. I added a screenshot test so it doesn't happen again.

**Comment by TobiasSuhrborg** [comment]
@shyndman Thanks a lot, this fixed my problem =)

**Comment by shyndman** [comment]
You've very welcome. Tell your friends! ;)

**Comment by TobiasSuhrborg** [comment]
Will do :)
